elastic-training/controllers/elastichorovodjob_controller.go

```diff
@@ -195,16 +296,20 @@ func (r *ElasticHorovodJobReconciler) desiredWorkers(ehjob aiv1alpha1.ElasticHor
 			Namespace: ehjob.Namespace,
 		},
 		Spec: appsv1.StatefulSetSpec{
-			ServiceName: svcName,
-			Replicas:    ehjob.Spec.WorkersSpec.TargetReplicas,
+			ServiceName:         svcName,
+			Replicas:            ehjob.Spec.WorkersSpec.TargetReplicas,
+			PodManagementPolicy: appsv1.PodManagementPolicyType(string(appsv1.ParallelPodManagement)),
 			Selector: &metav1.LabelSelector{
 				MatchLabels: map[string]string{"ElasticHorovodJob": ehjob.Name, "role": "worker"},
 			},
 			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{"ElasticHorovodJob": ehjob.Name, "role": "worker"},
+				ObjectMeta: metav1.ObjectMeta{ //add labels such as podgroup # and statefulset
+					Labels: map[string]string{"ElasticHorovodJob": ehjob.Name, "role": "worker",
+						"pod-group.scheduling.sigs.k8s.io": fmt.Sprintf("podgroup-ehjob-%s", ehjob.Name),
+					},
 				},
 				Spec: corev1.PodSpec{
+					SchedulerName: "default-scheduler",
```

Comment:
Is this necessary? I assume we are using the default scheduler anyway.

Reply:
I added default-scheduler for easier changing if the coscheduling scheduler is configured to another name; comment added